pypeit/images/scienceimage.py

```diff
@@ -45,6 +45,8 @@ class ScienceImage(pypeitimage.PypeItImage, maskimage.ImageMask):
             Inverse variance image
         rn2img (np.narray):
             Read noise**2 image
+        datasec_img (np.ndarray):
```

Comment:
I'm not following why the scienceimage needs to know about the datasec_img. The sciencimage should only operate on a processed gain multiplied image and this should take place in the call to process.

Reply:
This is a re-hash of Issues #740 and #719.  
As I mentioned in the closed PR, I started walking on this
path and found it too arduous for this PR.  Am not 
taking it on yet.